apps/cowswap-frontend/src/modules/swap/hooks/useEthFlowContext.ts

```diff
@@ -9,23 +8,18 @@ import { useTransactionAdder } from 'legacy/state/enhancedTransactions/hooks'
 
 import { FlowType, getFlowContext, useBaseFlowContextSetup } from 'modules/swap/hooks/useFlowContext'
 import { EthFlowContext } from 'modules/swap/services/types'
-import {
-  addInFlightOrderIdAtom,
-  ethFlowInFlightOrderIdsAtom,
-} from 'modules/swap/state/EthFlow/ethFlowInFlightOrderIdsAtom'
+import { addInFlightOrderIdAtom } from 'modules/swap/state/EthFlow/ethFlowInFlightOrderIdsAtom'
+
+import { useIsEthFlowOrderExists } from './useIsEthFlowOrderExists'
 
 export function useEthFlowContext(): EthFlowContext | null {
   const contract = useEthFlowContract()
   const baseProps = useBaseFlowContextSetup()
   const addTransaction = useTransactionAdder()
 
-  const ethFlowInFlightOrderIds = useAtomValue(ethFlowInFlightOrderIdsAtom)
   const addInFlightOrderId = useSetAtom(addInFlightOrderIdAtom)
 
-  const checkInFlightOrderIdExists = useCallback(
-    (orderId: string) => ethFlowInFlightOrderIds.includes(orderId),
-    [ethFlowInFlightOrderIds]
-  )
+  const isEthFlowOrderExists = useIsEthFlowOrderExists()
```

Comment:
The variable name indicates to me this is a boolean.
Reading the code though, it's a callback.

I suggest renaming the hook and callback to:

useCheckEthFlowOrderExists
checkEthFlowOrderExists

Reply:
Thanks! Renamed